Adding tagged base versioning and automatic build/deployment of wrims

jmdegeorge opened the pull request.

jmdegeorge committed:
Adding tagged base versioning and automatic build/deployment of wrims-gui installer on publish action.

jmdegeorge committed:
Updating GitHub token to one with REPO SCOPE for deploying artifact

jmdegeorge committed:
Adding "write-all" permission to the yaml action.

stephen-rma commented on .github/workflows/Java_CI.yaml:
```diff
@@ -30,8 +30,7 @@ jobs:
       run: |
         ./gradlew clean
         ./gradlew :wrims-core:build
```
wrims-core build should run when you call 'build' globally. Is there a reason to keep this line separate from the main build task?

jmdegeorge replied:
This has been fixed to no longer require an explicit build of wrims-core first.

stephen-rma commented on .github/workflows/build_installer.yaml:
```diff
@@ -0,0 +1,35 @@
+          PKG_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          ./gradlew clean
+          ./gradlew :wrims-core:build
```
Same feedback about wrims-core build vs global build here

jmdegeorge replied:
Fixed

stephen-rma commented on .github/workflows/build_installer.yaml:
```diff
@@ -0,0 +1,35 @@
+
+      - name: Attach WRIMS Installer zip to github deployment
+        run: |
+          gh release upload ${{ github.event.release.tag_name }} wrims-gui/build/installer/wrims_gui_x64_${{ github.event.release.tag_name }}.zip --repo ${{ github.repository }}
```
Is there an advantage to publishing it this way vs a Gradle Publish step that publishes to the GH Repo via Maven compat? That would allow doing `publishToMavenLocal` for local testing too, which is a benefit of doing it via Gradle... Then the publish step is just `gradlew publish` or similar, maybe passing in the repo path via a Gradle prop

stephen-rma replied:
This attaches it to a release vs just putting it in the package repo

stephen-rma commented on build.gradle:
```diff
@@ -18,6 +18,5 @@ def versionLabel(gitInfo) {
     group = 'gov.ca.dwr'
-    version = '1.0.0.0' //hard coding version to match wrims gui expectations.
-//    version = versionLabel(versionDetails())
+    version = versionLabel(versionDetails())
```
+1, this is good to see

stephen-rma committed:
Add explicit dependency of bundle plugin onto configurations

Bundle plugin does not properly declare dependency on Java configurations, so other projects do not generate jars before the bundle plugin tries to read them. The bundle plugin cannot read non-existent jars, so force an explicit dependency here so that the jars will be built before the Bundle process tries to read them.

jmdegeorge committed:
Cleaned up build.gradle files. Removed manual pre-build of wrims-core module. Fixed copyright info.

stephen-rma commented on wrims-core/build.gradle:
```diff
@@ -60,21 +62,6 @@ dependencies {
-    exclude group: "com.sun.media", module: "jai_imageio"
-}
-
-tasks.register('copyDependencies', Copy) {
```
Was this meant to be removed?

jmdegeorge replied:
Yes, this was a deliberate cleanup in the build.

stephen-rma reviewed (approved):
LGTM. Some good cleanup and DRY here, works around a dependency tracking issue from a 3rd party plugin, and switches to Tag versioning which is ideal.
